src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -331,18 +553,441 @@ namespace Opm {
+
+                    ijk[0] = i;
+                    ijk[1] = j;
+                    ijk[2] = k;
+
+                    getCellCorners(ijk, dims, X, Y, Z );
+
+                    m_volume[n] = calculateCellVol(X, Y, Z);
+
+                    std::array<double, 3> center;
+
+                    center[0]=std::accumulate(X.begin(), X.end(), 0.0) / 8.0;
+                    center[1]=std::accumulate(Y.begin(), Y.end(), 0.0) / 8.0;
+                    center[2]=std::accumulate(Z.begin(), Z.end(), 0.0) / 8.0;
+
+                    m_cellCenter[n] = center;
+
+                    // calculate dx
+
+                    double x2 = (X[1]+X[3]+X[5]+X[7])/4.0;
+                    double y2 = (Y[1]+Y[3]+Y[5]+Y[7])/4.0;
+
+                    double x1 = (X[0]+X[2]+X[4]+X[6])/4.0;
+                    double y1 = (Y[0]+Y[2]+Y[4]+Y[6])/4.0;
+
+                    double dx = sqrt(pow((x2-x1), 2.0) + pow((y2-y1), 2.0) );
+
+                    // calculate dy
+
+                    x2 = (X[2]+X[3]+X[6]+X[7])/4.0;
+                    y2 = (Y[2]+Y[3]+Y[6]+Y[7])/4.0;
+
+                    x1 = (X[0]+X[1]+X[4]+X[5])/4.0;
+                    y1 = (Y[0]+Y[1]+Y[4]+Y[5])/4.0;
+
+                    double dy = sqrt(pow((x2-x1), 2.0) + pow((y2-y1), 2.0));
+
+                    // calculate dz
+
+                    double z2 = (Z[4]+Z[5]+Z[6]+Z[7])/4.0;
+                    double z1 = (Z[0]+Z[1]+Z[2]+Z[3])/4.0;
+
+                    double dz = z2-z1;
+
+                    m_dx[n] = dx;
+                    m_dy[n] = dy;
+                    m_dz[n] = dz;
+
+                    m_depth[n] =  (z2 + z1) / 2.0;
+
+                    n++;
+                }
+            }
+        }
+    }
+
+    void EclipseGrid::getCellCorners(const std::array<int, 3>& ijk, const std::array<int, 3>& dims,
+                                     std::vector<double>& X,
+                                     std::vector<double>& Y,
+                                     std::vector<double>& Z) const
+    {
+
+        std::array<int, 8> zind;
+        std::array<int, 4> pind;
+        
+        if (X.size() < 8 || Y.size() < 8 || Z.size() < 8) {
+            OPM_THROW(std::invalid_argument,
+                      "In routine getCellCorners. X, Y and Z should be a vector of size 8");
+        }
+
+        // calculate indices for grid pillars in COORD arrray
+        const size_t p_offset = ijk[1]*(dims[0]+1)*6 + ijk[0]*6;
+        
+        pind[0] = p_offset;
+        pind[1] = p_offset + 6;
+        pind[2] = p_offset + (dims[0]+1)*6;
+        pind[3] = pind[2] + 6;
+
+        // get depths from zcorn array in ZCORN array
+        const size_t z_offset = ijk[2]*dims[0]*dims[1]*8 + ijk[1]*dims[0]*4 + ijk[0]*2;
+
+        zind[0] = z_offset;
+        zind[1] = z_offset + 1;
+        zind[2] = z_offset + dims[0]*2;
+        zind[3] = zind[2] + 1;
+
+        for (int n = 0; n < 4; n++) {
+            zind[n+4] = zind[n] + dims[0]*dims[1]*4;
+        }
```

Comment:
Ahhh - the braces spread; it is a disease!

Reply:
fixed